src/utils/apiFetch.ts

```diff
@@ -0,0 +1,19 @@
+import apiUrl from './apiUrl';
+
+export interface RequestWithHeaders {
+    headers: Record<string, string>;
+}
+
+export type ApiFetch = (path: string, init?: RequestInit) => Promise<Response>;
+
+export const createApiFetch = (req: RequestWithHeaders): ApiFetch => {
```

Comment:
I just realized that since the only thing we use off the `req` object is it's headers, we could actually just take in the `headers` here (typed as `Record<string,string>`) and reduce code significantly. That takes away the need for a special interface exported from this file, and it takes away the need to cast to that special interface in the places where `createApiFetch()` is used.